CRM/CiviAwards/Form/AwardReview.php

```diff
@@ -63,11 +63,21 @@ class CRM_CiviAwards_Form_AwardReview extends CRM_Core_Form {
    * {@inheritDoc}
    */
   public function setDefaultValues() {
-    if (empty($this->defaultValues) && $this->activityId) {
-      $this->defaultValues = $this->getProfileFieldValues();
-
-      return $this->defaultValues;
+    $this->defaultValues = [];
+    $isAddAction = $this->_action & CRM_Core_Action::ADD;
+
+    if ($this->activityId) {
```

Comment:
Any reason why we can't have the original if condition `if (empty($this->defaultValues) && $this->activityId)`.
The reason is that the preprocess might get called a couple of time (e.g when submitting) and no need to set the default values again if it is already set.

Reply:
I was not aware of this!

Since we need to set some values depending if the form is in editing mode and some others for create mode, I have made it so that the function is finished immediately to avoid setting the values again.